packages/components/src/List/ListComposition/SortBy/SortBy.component.js

```diff
@@ -40,7 +40,7 @@ function SortBy(props) {
 	const onOrderChange = event =>
 		performChange(event, { ...currentValue, isDescending: !currentValue.isDescending });
 
-	return (
+	return displayMode !== 'table' ? (
```

Comment:
Can you avoid the ternary and use a fast return ?
```javascript
if (displayMode === 'table') { // or a constant as suggested by frassinier
    return null;
}

return ...
```

Reply:
done